### ISSUE TITLE
Potential core  hack for  autoloader

### ISSUE THREAD

**Issue by eileenmcnaughton** [prondubuisi/psalmoutput#1]
@prondubuisi @jgleeson  - perhaps this will load the CiviCRM_API3_Exception file - it's not  loading due to weird naming. That isn't an issue in core I suspect as the file winds up being otherwise loaded because API_Exception is probably always called first (that being the internal exception which leads to the throwing of external exception CiviCRM_API3_Exception)

```
--- a/CRM/Core/ClassLoader.php
+++ b/CRM/Core/ClassLoader.php
@@ -205,6 +205,9 @@ class CRM_Core_ClassLoader {
    * @param $class
    */
   public function loadClass($class) {
+    if ($class === 'CiviCRM_API3_Exception') {
+      require_once 'api/Exception.php';
+    }
```

**Comment by jackgleeson** [comment]
I tested this fix with psalm only scanning api/v3/examples, and it removed all UndefinedClass errors taking the error count down from 310 to 16.

**Comment by eileenmcnaughton** [comment]
@prondubuisi if this works for you please create a PR against the CiviCRM repo

**Comment by prondubuisi** [comment]
> I tested this fix with psalm only scanning api/v3/examples, and it removed all UndefinedClass errors taking the error count down from 310 to 16.

@jackgleeson How and where did you implement the fix @eileenmcnaughton any heads up please?

**Comment by eileenmcnaughton** [comment]
@prondubuisi it's in the core codebase - are you  familiar with reading patches? https://ramblings.narrabilis.com/generating-and-reading-patch-files

**Comment by prondubuisi** [comment]
> @prondubuisi it's in the core codebase - are you familiar with reading patches? https://ramblings.narrabilis.com/generating-and-reading-patch-files

Thanks for sharing, I can't seem to understand it. Am not familiar with patches

**Comment by eileenmcnaughton** [comment]
@prondubuisi understanding patches is an important skill - I'd take the time to sit down & read through something like this

https://www.pair.com/support/kb/paircloud-diff-and-patch/

or this one

https://www.poftut.com/patch-command-tutorial-with-examples-for-linux/

**Comment by prondubuisi** [comment]
Figured it out @eileenmcnaughton @jackgleeson You mentioned I can create a pull request and then submit a patch right?

**Comment by eileenmcnaughton** [comment]
Yes! See how far this gets you https://docs.civicrm.org/dev/en/latest/tools/git/#pr

**Comment by prondubuisi** [comment]
Thanks

**Comment by prondubuisi** [comment]
Hello @eileenmcnaughton @jackgleeson here is a link to the pull request for this https://github.com/civicrm/civicrm-core/pull/15043
